cookbook/src/tx_segwit-v0.md

```diff
@@ -0,0 +1,342 @@
+- `script_code` is the script code required to spend a P2WPKH output; it is a reference to [`Script`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/script/struct.Script.html) type.
+   We are using the `script_code` variable that we defined earlier.
+- `value` is the amount of the UTXO we are spending; it is a [`u64`](https://doc.rust-lang.org/std/primitive.u64.html) type.
+   We are using the `const DUMMY_UTXO_AMOUNT` that we defined earlier.
+- `sighash_type` is the type of sighash; it is a [`EcdsaSighashType`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html) enum.
+   We are using the [`All`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html#variant.All) variant,
+   that sign all the outputs.
+
+We compute the message `msg` by converting the `sighash` to a [`Message`](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html) type.
+This is a hashed message input to an ECDSA signature.
+The [Message::from](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html#impl-From%3C%26%27_%20bitcoin%3A%3Ahashes%3A%3Asha256d%3A%3AHash%3E) method takes anything that implements the [`Hash` trait](https://docs.rs/bitcoin_hashes/latest/bitcoin_hashes/trait.Hash.html) as an argument[^ThirtyTwoByteHash].
+
+We compute the signature `sig` by using the [`sign_ecdsa`](https://docs.rs/secp256k1/latest/secp256k1/struct.Secp256k1.html#method.sign_ecdsa) method.
+It takes a refence to a [`Message`](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html) and a reference to a [`SecretKey`](https://docs.rs/secp256k1/latest/secp256k1/struct.SecretKey.html) as arguments,
+and returns a [`Signature`](https://docs.rs/secp256k1/latest/secp256k1/ecdsa/struct.Signature.html) type.
+
+In the next step, we update the `sighash_cache` first with [`witness_mut`](https://docs.rs/bitcoin/latest/bitcoin/sighash/struct.SighashCache.html#method.witness_mut) method.
+This allows us to modify the witness stack of the input we are signing.
+It returns an [`Option<&mut Witness>`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/witness/struct.Witness.html) type.
+Next, the witness stack is by using the [`push_bitcoin_signature`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/witness/struct.Witness.html#method.push_bitcoin_signature) method.
+It expects two arguments:
+
+1. A reference to a [`SerializedSignature`](https://docs.rs/secp256k1/0.27.0/secp256k1/ecdsa/serialized_signature/struct.SerializedSignature.html) type.
+   This is accomplished by calling the [`serialize_der`](https://docs.rs/secp256k1/latest/secp256k1/ecdsa/struct.Signature.html#method.serialize_der) method on the `Signature` `sig`,
+   which returns a `SerializedSignature` type.
+1. A [`EcdsaSighashType`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html) enum.
+   Again we are using the same [`All`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html#variant.All) variant that we used earlier.
+
```

Comment:
Here we need to push the public key as well (this was a bug in my solution to the workshop found by @sanket1729, an attempt to improve the API is made here: https://github.com/rust-bitcoin/rust-bitcoin/pull/2084

Reply:
Ok I am following the PR. From my understanding, this does not change the code we will show in the cookbook.

Reply:
The code is missing the push of the pubkey, I commented on the line of code where its needed. The PR I linked to adds a function so that one does not have to push the sig and keys separately but it won't be there till next release.

Reply:
fixed